src/appleseed.shaders/src/max/as_max_blend_material.osl

```diff
@@ -0,0 +1,90 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2017 Sergo Pogosyan, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#include "as_osl_extensions.h"
+
+surface as_max_blend_material
+(
+    closure color BaseMtl = 0,
+    
+    closure color LayerMtl_1 = 0,
+    closure color LayerMtl_2 = 0,
+    closure color LayerMtl_3 = 0,
+    closure color LayerMtl_4 = 0,
+    closure color LayerMtl_5 = 0,
+    closure color LayerMtl_6 = 0,
+    closure color LayerMtl_7 = 0,
+    closure color LayerMtl_8 = 0,
+    closure color LayerMtl_9 = 0,
+
+    color	MixColor_1 = color(0),
+    color	MixColor_2 = color(0),
+    color	MixColor_3 = color(0),
+    color	MixColor_4 = color(0),
+    color	MixColor_5 = color(0),
+    color	MixColor_6 = color(0),
+    color	MixColor_7 = color(0),
+    color	MixColor_8 = color(0),
+    color	MixColor_9 = color(0),
+    
+    float	MaskColor_1 = 1,    // Mask texture. Set to 1 to not zero out a layer in case when no texture is connected
+    float	MaskColor_2 = 1,
+    float	MaskColor_3 = 1,
+    float	MaskColor_4 = 1,
+    float	MaskColor_5 = 1,
+    float	MaskColor_6 = 1,
+    float	MaskColor_7 = 1,
+    float	MaskColor_8 = 1,
+    float	MaskColor_9 = 1,
+
+    output	closure color ClosureOut = 0
+)
+{
+    ClosureOut = BaseMtl;
+    
+    if (LayerMtl_1)
+        ClosureOut = mix(ClosureOut, LayerMtl_1, (MixColor_1 * MaskColor_1));
+    if (LayerMtl_2)
+        ClosureOut = mix(ClosureOut, LayerMtl_2, (MixColor_2 * MaskColor_2));
```

Comment:
Is this the right mixing algorithm? Or more precisely, the mixing algorithm we actually want?

Here, the weight of `LayerMtl_1` will be

    (MixColor_1 * MaskColor_1) * (1 - MixColor_2 * MaskColor_2)

It's likely that this is what we want. Just checking.